### ISSUE TITLE
Model only generates <eos> tokens, Text Generation,

### ISSUE THREAD

**Issue by patil-suraj** [lucidrains/reformer-pytorch#41]
Greetings! I've found this repo very useful,flexible and easy to use. Thanks for putting it out. I've been playing with this repo for a text generation problem.

I want to generate a reply given previous history of conversation. Here's how I'm encoding the sequence.
`<bos><speaker1>Hello, how are you?<speaker2>Great! What about you?<eos>`.
As can be seen from the enwiki8 example, while giving input we need to drop the last token from the sequence and for target we start from 2nd token to last token. So for above example
`inp: <bos><speaker1>Hello, how are you?<speaker2>Great! What about you?`
`targets: <speaker1>Hello, how are you?<speaker2>Great! What about you?<eos>`
I'm calculating loss only on the last portion i.e on `Great! What about you?<eos>`.

This is unlike other models, i.e with GPT-2 or Trax implementation of Reformer where you just feed the same sequence as input and targets, and it handles the rest.
So when I trained the model with above encoding, the model only generates `<eos>` tokens. So I removed the `<eos>` token and trained again but then the last token was always some punctuation, so the model was only generating punctuation.

Is this really an issue or am I doing something wrong ? Also can we make it more like the Trax implementation where we just feed the same sequence for both input and targets ?

**Comment by lucidrains** [comment]
@patil-suraj Hi Patil! Thanks for your interest. Could you possibly link me to a gist of the training code you are using?

**Comment by lucidrains** [comment]
@patil-suraj I see how they do it. I can do something similar where I have a wrapper that just computes the cross entropy loss between adjacent tokens, and you can just feed a single sequence in

**Comment by lucidrains** [comment]
@patil-suraj I added a helper tool here https://github.com/lucidrains/reformer-pytorch#helpers-for-training-auto-regressive-models Hopefully that makes it easier for you to train

**Comment by patil-suraj** [comment]
@lucidrains Thank you for the quick response!. Will try it out.

**Comment by patil-suraj** [comment]
@patil-suraj  I see that the model now calculates the loss itself, is there any way I can pass a loss mask ? I only need to compute loss on the last portion of sequence. Thanks in advance!
